File: DSA/DSA03034 DÃY CON CHUNG.cpp
#include <bits/stdc++.h>
using namespace std;

void submit()
{
    int x, y, z, i = 0, j = 0, k = 0;
    cin >> x >> y >> z;

    vector<long long> A(x), B(y), C(z);
    vector<long long> out;
    for (auto &&i : A)  cin >> i;
    for (auto &&i : B)  cin >> i;
    for (auto &&i : C)  cin >> i;

    bool check = false;
    while (i < x && j < y && k < z)
    {
        if (A[i] == B[j] && B[j] == C[k])
        {
            out.push_back(A[i]);
            check = true;
            i++, j++, k++;
        }
        else if (A[i] > B[j])
        {
            j++;
        }
        else if (B[j] > C[k])
        {
            k++;
        }
        else
        {
            i++;
        }
    }
    
    if (check)
    {
        for (auto &&i : out)
        {
            cout << i << " ";
        }
    }
    else
    {
        cout << "NO";
    }
    

    cout << endl;
}

int main()
{
    ios::sync_with_stdio(false);                                    // Speed up input/output
    cin.tie(nullptr);
    cout.tie(nullptr);

    int test;
    cin >> test;
    while (test--)
    {
        submit();
    }
    

    return 0;
}

File: DSA_python/DSA03034 DÃY CON CHUNG.py
def submit():
    x, y, z = map(int, input().split())
    i, j, k = 0, 0, 0

    A = list(map(int, input().split()))
    B = list(map(int, input().split()))
    C = list(map(int, input().split()))
    out = []
    check = False

    while i < x and j < y and k < z:
        if A[i] == B[j] == C[k]:
            out.append(A[i])
            check = True
            i += 1; j += 1; k += 1
        elif A[i] > B[j]:
            j += 1
        elif B[j] > C[k]:
            k += 1
        else:
            i += 1

    print(" ".join(map(str, out)) if check == True else "NO")


test = int(input())
for i in range(test):
    submit()